vcr/credential/validator.go

```diff
@@ -123,3 +123,87 @@ func (d nutsOrganizationCredentialValidator) Validate(credential vc.VerifiableCr
 
 	return nil
 }
+
+// nutsAuthorizationCredentialValidator checks for mandatory fields: id, legalBase, purposeOfUse.
+// It checks if the value for legalBase.consentType is either 'explicit' or 'implied'.
+// When 'explicit', both the evidence and subject subfields must be filled.
+// When 'implied', the restrictions array must have values.
+type nutsAuthorizationCredentialValidator struct{}
+
+func (d nutsAuthorizationCredentialValidator) Validate(credential vc.VerifiableCredential) error {
+	var target = make([]NutsAuthorizationCredentialSubject, 0)
+
+	if err := validate(credential); err != nil {
+		return err
+	}
+
+	if !credential.IsType(*NutsAuthorizationCredentialTypeURI) {
+		return failure(fmt.Sprintf("type '%s' is required", NutsAuthorizationCredentialType))
+	}
+
+	// if it fails, length check will trigger
+	_ = credential.UnmarshalCredentialSubject(&target)
+	if len(target) != 1 {
+		return failure("single CredentialSubject expected")
+	}
+	cs := target[0]
+
+	if cs.ID == "" {
+		return failure("'credentialSubject.ID' is nil")
+	}
+	if cs.PurposeOfUse == "" {
+		return failure("'credentialSubject.PurposeOfUse' is nil")
+	}
+	switch cs.LegalBase.ConsentType {
+	case "implied":
+		if len(cs.Restrictions) == 0 {
+			return failure("'credentialSubject.Restrictions' must have entries when consentType is 'implied'")
```

Comment:
consistency; below you refer to `Restriction[]` (with added brackets)

Reply:
Added `[]` everywhere